### PR TITLE
Set HTTP parameter include_type_name=true if ES is >=7.0

### PR DESCRIPTION
**Pull request by urso**
With removal of types from ES 7.0, we will set include_type_name=true so
templates can still be installed correctly.

Companion PR for 7.0 is #10156.

Both PRs should be merged after both are green and I had a chance to verify upgrading/installing between a few Beats/ES combinations works seamlessly.

### CODE REVIEW

File: libbeat/outputs/elasticsearch/client.go [hunk 1]
```diff
@@ -312,7 +308,15 @@ func (client *Client) publishEvents(
 		return nil, nil
 	}
 
-	requ := client.bulkRequ
+	params := mergeParams(client.params, VersionParams(client.GetVersion(), true))
+
+	// requ := client.bulkRequ
```
Comment (ruflin) [inline libbeat/outputs/elasticsearch/client.go]: Leftover?

**Comment by ruflin** [comment]
This seems to solve the template loading issue but afterwards I get:

```
2019-01-18T11:16:10.048+0100	WARN	elasticsearch/client.go:523	Cannot index event publisher.Event{Content:beat.Event{Timestamp:time.Time{wall:0xbf0887b9587a9618, ext:3045274864, loc:(*time.Location)(0x6b2af60)}, Meta:common.MapStr(nil), Fields:common.MapStr{"metricset":common.MapStr{"module":"system", "rtt":6510, "name":"fsstat"}, "event":common.MapStr{"dataset":"system.fsstat", "duration":6510892}, "system":common.MapStr{"fsstat":common.MapStr{"total_size":common.MapStr{"free":0x1f28e4f4000, "used":0x21931ffe200, "total":0x40bc04f2200}, "count":7, "total_files":0x800000002ae88307}}, "beat":common.MapStr{"name":"xodoa.local", "hostname":"xodoa.local", "version":"6.7.0"}, "host":common.MapStr{"architecture":"x86_64", "name":"xodoa.local", "os":common.MapStr{"platform":"darwin", "version":"10.14.2", "family":"darwin", "name":"Mac OS X", "build":"18C54"}, "id":"DF70DAEB-86C4-57A9-A109-941526B0056B"}}, Private:interface {}(nil)}, Flags:0x0} (status=400): {"type":"mapper_parsing_exception","reason":"failed to parse field [system.fsstat.total_files] of type [long]","caused_by":{"type":"json_parse_exception","reason":"Numeric value (9223372037574656775) out of range of long (-9223372036854775808 - 9223372036854775807)\n at [Source: org.elasticsearch.common.bytes.BytesReference$MarkSupportingStreamInputWrapper@59a8f883; line: 1, column: 184]"}}
```

**Comment by urso** [comment]
The error is pretty weird. Which ES version are you using? It looks like the template was not applied to the index.

**Comment by urso** [comment]
Testing with Elasticsearch master and this PR I have no problems with metricbeat ingesting long values.

But I noticed: `"total_files":0x800000002ae88307`
Unfortunately the log doesn't tell which type it is, but bit 63 is set. I assume its an uint64 in metricbeat.

This also matches the error message: `"Numeric value (9223372037574656775) out of range of long (-9223372036854775808 - 9223372036854775807)`

This value looks somewhat funny. Are you sure the number is correct? Does bit 63 potentially have a different meaning (a flag)?

The 63bit masking is only available in master, as this was supposed to be a breaking change. This is why indexing with this value fails.

**Comment by urso** [comment]
Jenkins, test this

**Comment by ruflin** [comment]
I tested this again but disabling the `fsstat` metricset. Now things seem to work as expected. 

@jsoriano I tested this on my normal OS X machine with Mojave. Does this mean fsstat does should be disabled on OS X in MB or could we fix this specific issue?

**Comment by urso** [comment]
> I tested this on my normal OS X machine with Mojave. Does this mean fsstat does should be disabled on OS X in MB or could we fix this specific issue?

I'm testing on Mojave as well :)
Do you have any other mounts in your system (USB, Network, ...) ?

**Comment by ruflin** [comment]
Right now I have nothing else mounted :-( I'm on 10.14.2.

**Pull request merged**